js/main.js

```diff
@@ -71,3 +71,12 @@ const customSoftwarePopoverList = [...document.querySelectorAll('.custom-softwar
   title: 'Custom Software 2FA'
 }));
 
+const backToTopElement = $('.back-to-top-link');
+backToTopElement.removeClass('invisible');
+$(window).scroll(() => {
```

Comment:
I'm not very good at JS but instinctively feel like a function like this should be asynchronous. Would there be any reason for it not to?

Reply:
I'm also not good at JS, so I tested it with async and it did the same job.

Reply:
I'm, not sure that adding the `async` keyword will do anything to this function. `await` isn't used inside this function, so using `async` will just cause this function to return a promise. I think that even with `async`, this code will still run synchronously. See https://stackoverflow.com/a/51201849

Also, IMHO it's not worth hiding the back to top button for users without JS as the site is unusable without it anyway and we have the separate noscript page for non-JS users. Other than that, looks good 👍!

Reply:
@ApCoder123 thanks for helping with the JS. I applied your suggestions.
Little side notice around the `async` topic. Could you please check these lines/functions?

- [main.js#20](https://github.com/2factorauth/twofactorauth/blob/ba52fb5b52450dd0de376b20c6ee3362ce9bbb98/js/main.js#L20)
- [main.js#31](https://github.com/2factorauth/twofactorauth/blob/ba52fb5b52450dd0de376b20c6ee3362ce9bbb98/js/main.js#L31)

Reply:
Yeah, I'm pretty sure that those uses of `async` will have no effect on the execution of the function